cmd/scheduler/main.go

```diff
@@ -0,0 +1,192 @@
+	}
+	return nil
+}
+
+func gatherOptions(fs *flag.FlagSet, args ...string) options {
+	logrusutil.ComponentInit("lighthouse-scheduler")
+
+	var o options
+	fs.StringVar(&o.namespace, "namespace", "", "The namespace to create cronjobs in")
+	fs.StringVar(&o.configPath, "config-path", "", "Path to config.yaml.")
+	fs.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
+
+	err := fs.Parse(args)
+	if err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
+	}
+
+	return o
+}
+
+func main() {
+	o := gatherOptions(flag.NewFlagSet(os.Args[0], flag.ExitOnError), os.Args[1:]...)
+	if err := o.Validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
+	}
+	lhConfig, err := config.Load(o.configPath, o.jobConfigPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("error starting config map watcher")
+	}
+	if err := updateCronjobs(o.namespace, lhConfig.JobConfig); err != nil {
+		logrus.WithError(err).Fatal("Could not update cron jobs")
+	}
+}
+
+func updateCronjobs(namespace string, jobConfig job.Config) error {
+	cfg, err := clients.GetConfig("", "")
+	if err != nil {
+		return err
+	}
+	k8sclient, err := kubeclient.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	observed := map[string]v1beta1.CronJob{}
+	actual, err := k8sclient.BatchV1beta1().CronJobs(namespace).List(metav1.ListOptions{LabelSelector: periodicSelector})
+	if err != nil {
+		return err
+	}
+	for _, cronjob := range actual.Items {
+		observed[cronjob.Name] = cronjob
+	}
+	for _, periodic := range jobConfig.Periodics {
+		cronjob, err := makeCronJob(namespace, periodic)
+		if err != nil {
+			return err
+		}
+		if _, ok := observed[cronjob.Name]; !ok {
+			// create
+			_, err := k8sclient.BatchV1beta1().CronJobs(namespace).Create(cronjob)
+			if err != nil {
+				return err
+			}
+		} else {
+			// update
+			_, err := k8sclient.BatchV1beta1().CronJobs(namespace).Update(cronjob)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	lookup := func(cronjob v1beta1.CronJob) bool {
+		for _, periodic := range jobConfig.Periodics {
+			if cronjob.Name == periodic.Name {
+				return true
+			}
+		}
+		return false
+	}
+	for _, cronjob := range observed {
+		if !lookup(cronjob) {
+			// delete
+			err := k8sclient.BatchV1beta1().CronJobs(namespace).Delete(cronjob.Name, nil)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func makeCronJob(namespace string, periodic job.Periodic) (*v1beta1.CronJob, error) {
+	var one int32 = 1
+	var true bool = true
+	lhjob := lighthousev1alpha1.LighthouseJob{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: periodic.Name + "-",
+			Namespace:    namespace,
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion:         "batch/v1beta1",
+				Kind:               "CronJob",
+				Name:               periodic.Name,
+				Controller:         &true,
+				BlockOwnerDeletion: &true,
+			}},
+		},
+		Spec: jobutil.PeriodicSpec(periodic),
+	}
+	b, err := yaml.Marshal(lhjob)
```

Comment:
The lh job will probably need to be serialized into json so that it's a single line.